From c2e084db84de1d0f1f413ebd6255b19e033c34d3 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 20 May 2020 16:02:27 +0200
Subject: [PATCH 1/2] [addons][filesystem] fix wrongly used -1 as return on
 bool

Has added there a wrong return in case of not available file handle.
---
 xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h
index 6664e2be196d..385c2d49ce64 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h
@@ -2115,7 +2115,7 @@ class CFile
     using namespace kodi::addon;
 
     if (!m_file)
-      return -1;
+      return false;
     return CAddonBase::m_interface->toKodi->kodi_filesystem->io_control_get_seek_possible(
         CAddonBase::m_interface->toKodi->kodiBase, m_file);
   }
@@ -2136,7 +2136,7 @@ class CFile
     using namespace kodi::addon;
 
     if (!m_file)
-      return -1;
+      return false;
     return CAddonBase::m_interface->toKodi->kodi_filesystem->io_control_get_cache_status(
         CAddonBase::m_interface->toKodi->kodiBase, m_file, status);
   }
@@ -2154,7 +2154,7 @@ class CFile
     using namespace kodi::addon;
 
     if (!m_file)
-      return -1;
+      return false;
     return CAddonBase::m_interface->toKodi->kodi_filesystem->io_control_set_cache_rate(
         CAddonBase::m_interface->toKodi->kodiBase, m_file, rate);
   }
@@ -2172,7 +2172,7 @@ class CFile
     using namespace kodi::addon;
 
     if (!m_file)
-      return -1;
+      return false;
     return CAddonBase::m_interface->toKodi->kodi_filesystem->io_control_set_retry(
         CAddonBase::m_interface->toKodi->kodiBase, m_file, retry);
   }

From 9f824c6e93a0ae214acb93cde131950e344d773f Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 20 May 2020 16:04:19 +0200
Subject: [PATCH 2/2] [addons] increase filesystem API version to 1.1.1

Min version stays at 1.1.0 and no addon need update.
---
 xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
index fcb0b0438549..ad3aac0b0e15 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
@@ -60,7 +60,7 @@
 #define ADDON_GLOBAL_VERSION_AUDIOENGINE_DEPENDS      "AudioEngine.h" \
                                                       "c-api/audio_engine.h"
 
-#define ADDON_GLOBAL_VERSION_FILESYSTEM               "1.1.0"
+#define ADDON_GLOBAL_VERSION_FILESYSTEM               "1.1.1"
 #define ADDON_GLOBAL_VERSION_FILESYSTEM_MIN           "1.1.0"
 #define ADDON_GLOBAL_VERSION_FILESYSTEM_XML_ID        "kodi.binary.global.filesystem"
 #define ADDON_GLOBAL_VERSION_FILESYSTEM_DEPENDS       "Filesystem.h" \
